crates/shim-sgx/src/handler/mod.rs

```diff
@@ -109,6 +104,108 @@ impl<'a> Handler<'a> {
                 .unwrap_or_else(|_| self.attacked());
         }
     }
+
+    fn mprotect_unlocked(
+        &mut self,
+        heap: &mut Heap,
+        addr_in: NonNull<c_void>,
```

Comment:
```suggestion
        addr: NonNull<c_void>,
```

Reply:
Then, `let addr = Address::new(addr);` would mask i:
```
error[E0308]: mismatched types
    --> crates/shim-sgx/src/handler/mod.rs:139:28
     |
139  |         self.mprotect_host(addr, length.bytes(), prot)?;
     |              ------------- ^^^^ expected struct `NonNull`, found struct `primordial::Address`
     |              |
     |              arguments to this function are incorrect
     |
     = note: expected struct `NonNull<c_void>`
                found struct `primordial::Address<usize, primordial::Page>`
```

Reply:
It also IMHO goes out of the scope of moving the code. I would like to keep it intact as possible when I move a code block.